imager/imager/urls.py

```diff
@@ -13,26 +13,58 @@
     1. Import the include() function: from django.conf.urls import url, include
     2. Add a URL to urlpatterns:  url(r'^blog/', include('blog.urls'))
 """
-from django.conf.urls import url
+from django.conf.urls import url, include
 from django.contrib import admin
-from .views import home_page
-from .views import ClassView
-# from django.conf.urls.static import static
-# from imager import settings
+from .views import ClassView, logout_view, LibraryView
+from .views import AlbumView, AddAlbumView, EditAlbumView, DeleteAlbumView
+from .views import PhotoView, AddPhotoView, EditPhotoView, DeletePhotoView
+from .views import ProfileView, ProfileEditView
+from django.conf.urls.static import static
+from imager import settings
+from django.contrib.auth.decorators import login_required
 
-# image_urls = []
-# profile_urls = []
-# api_urls = []
-#
-# urlpatterns = image_urls + profile_urls + api_urls
 
 urlpatterns = [
     url(r'^admin/', admin.site.urls),
-    url(r'^home/(?P<id>[0-9]+)$', ClassView.as_view(), name='home_page'),
-    # url(r'^home/(?P<id>[0-9]+)$', home_page, name='home_page'),
-    # url(r'^profile/', imager_profile.urls),
+    url(r'^$', ClassView.as_view(), name='home_page'),
+    # url(r'^login/$', 'django.contrib.auth.views.login'),
+    url(r'^logout/$', logout_view),
+    url(r'^accounts/', include('registration.backends.hmac.urls')),
+    url(r'^accounts/profile/$',
+        login_required(ProfileView.as_view()),
+        name='profile'),
+    url(r'^images/library/$',
+        login_required(LibraryView.as_view()),
+        name='library'),
+    url(r'^images/album/(?P<id>[0-9]+)$',
+        login_required(AlbumView.as_view()),
+        name='album'),
+    url(r'^images/albums/add/$',
+        login_required(AddAlbumView.as_view()),
+        name='add_album'),
+    url(r'^images/albums/edit/(?P<id>[0-9]+)$',
+        login_required(EditAlbumView.as_view()),
+        name='edit_album'),
+    url(r'^images/albums/delete/(?P<id>[0-9]+)$',
+        login_required(DeleteAlbumView.as_view()),
+        name='delete_album'),
+    url(r'^images/photo/(?P<id>[0-9]+)$',
+        login_required(PhotoView.as_view()),
+        name='photo'),
+    url(r'^images/photos/add/$',
+        login_required(AddPhotoView.as_view()),
+        name='add_photo'),
+    url(r'^images/photos/edit/(?P<id>[0-9]+)$',
+        login_required(EditPhotoView.as_view()),
+        name='edit_photo'),
+    url(r'^images/photos/delete/(?P<id>[0-9]+)$',
+        login_required(DeletePhotoView.as_view()),
+        name='delete_photo'),
+    url(r'^profile/edit/$',
+        login_required(ProfileEditView.as_view()),
+        name='profile_edit'),
 ]
```

Comment:
It's better Django practice to shard these URLs out into the `urls.py` of their respective paths. In this `urls.py`, then, you would want to do something like this:
```
url(r'^images/', include('imager_images.urls'))
```

And then in `imager_images/urls.py`, something like this:
```
url(r'^album/(?P<id>[0-9]+)$',
    login_required(AlbumView.as_view()),
    name='album')
```